### DISCUSSION TITLE
Problem with conditonal rendering/data fetching on User Input

### DISCUSSION THREAD

**Discussion by mzschunke** [neuefische/questions#614] [Web] [answered]
Hi all! 

I am currently working on my [US7](https://github.com/mzschunke/capstone-project/issues/17) (Compare function)

Right now, I am really struggling with implementing the fetch/conditional rendering of an additional player. I've tried a lot of different things and feel kind of stuck now regarding my code. 

Status Quo: Rendering of one player (1 or 2) works fine, however after selecting the other one, I either get an error (playerStats are undefined) or cannot select another player. 

Sadly I cannot findout what's missing in this logic...

I've just pushed my WIP [click here](https://github.com/mzschunke/capstone-project/tree/develop/compare-function)

Your advice is highly aprpeciated!

**Comment by mbosselmann** [comment, marked as answer]
Moritz and I rebuilt both the stats page and the stats display component to make the logic work again. We made the fetching of the data for each player independent from each other and instead of using if else statements we used conditional rendering to have less repetitions in the code.

**Reply by mzschunke** [reply]
Thank you Mareike for taking the time to look into this!
Should be working as intended now. :)
